torch/csrc/jit/passes/dead_code_elimination.cpp

```diff
@@ -205,7 +205,11 @@ class DeadCodeEliminator {
           sweep(block, true);
         }
       }
-      if (!marked_.count(node)) {
+      // NB: Checking hasUses() is required. AD graphs are not perfectly
```

Comment:
Future PR should change AD graphs so that they are valid graphs and remove this workaround. (just put the unattached block into the forward graph temporarily).